plugins/mysql.yaml

```diff
@@ -87,7 +87,7 @@ pipeline:
 
   - id: slow_query_regex_parser
     type: regex_parser
-    regex: '# Time: (?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+Z)\s# User@Host:\s+(?P<dbuser>\w+)\[(?P<dbname>\w+)\]\s+@\s+\[(?P<ip_address>[\d\.]+)\]\s+Id:\s+(?P<id>\d+)\s+#\s+Query_time:\s+(?P<query_time>[\d\.]+)\s+Lock_time:\s+(?P<lock_time>[\d\.]+)\s+Rows_sent:\s+(?P<rows_sent>\d+)\s+Rows_examined:\s(?P<rows_examined>\d+)\s+(?P<query>(?s).*[^\s])'
+    regex: '# Time: (?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+Z)\s# User@Host:\s+(?P<dbuser>\w+)\[(?P<dbname>\w+)\]\s+@\s+((?P<host>[^\s]+)\s)?\[(?P<ip_address>[\d\.]+|)\]\s+Id:\s+(?P<id>\d+)\s+#\s+Query_time:\s+(?P<query_time>[\d\.]+)\s+Lock_time:\s+(?P<lock_time>[\d\.]+)\s+Rows_sent:\s+(?P<rows_sent>\d+)\s+Rows_examined:\s(?P<rows_examined>\d+)\s+(?P<query>(?s).*[^\s])'
```

Comment:
I believe this will still capture that group -- can you check that we're not getting an extra empty field because of this?

Reply:
You might need to do a non-capturing group `(?:a*)`

Reply:
It does create an empty field. Anytime there is a group in the regex it always captures the field empty or not. I don't see an issue with an empty field, but we structure it if it has an empty field.

Reply:
@ericwholt That non-capturing group construct should avoid making an empty field. Instead of just using parenthesis, add a `?:` inside them, and it should fix it

Reply:
See https://github.com/google/re2/wiki/Syntax for details

Reply:
Looks like unnamed capture groups are ignored anyways, so this is fine: https://github.com/observIQ/stanza/blob/9ba5650ee40dccc474c03391eaa3c0c5edbad48e/operator/builtin/parser/regex/regex.go#L109-L111